[bullet3] build failure Mac M1 / arm64

**Host Environment**
Mac M1 / arm64
Apple clang version 13.0.0 (clang-1300.0.29.3)
Target: arm64-apple-darwin21.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**To Reproduce**
`./vcpkg install bullet3`

**Failure logs** 
  package: bullet3[core]:arm64-osx -> 3.17#2
  vcpkg version: 2021-10-25-b4bff9afacafb0a234a8f20fd6451b2362c0f9a0
  vcpkg-tool version: 5bdb9d601 2021-10-29 (28 hours ago)

Starting package 1/1: bullet3:arm64-osx
Building package bullet3[core]:arm64-osx...
-- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/jmacey/vcpkg/triplets/community/arm64-osx.cmake
CMake Error at scripts/cmake/vcpkg_fail_port_install.cmake:96 (message):
  Architecture 'arm64' not supported by bullet3!

If I remove the following 

#vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "UWP")

from ports/bullet3/portfile.cmake

It works so I guess the check on arm64 and UWP is causing the issue. (I guess UWP is Universal Windows Platform?).

@jmacey On arm64-windows build:
```
[1/182] "C:\PROGRA~2\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\Hostx64\arm64\cl.exe"   /TP -DNO_OPENGL3 -DUSE_GRAPHICAL_BENCHMARK -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_WIN64 -IF:\vcpkg\buildtrees\bullet3\src\3.17-e90f5ef08b.clean\src /GR- /Gy /GF  /nologo /DWIN32  /W3 /utf-8  /EHsc /MP  /fp:fast /wd4244 /wd4267 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -D_DEBUG /showIncludes /Fosrc\BulletInverseDynamics\CMakeFiles\BulletInverseDynamics.dir\IDMath.cpp.obj /Fdsrc\BulletInverseDynamics\CMakeFiles\BulletInverseDynamics.dir\BulletInverseDynamics.pdb /FS -c F:\vcpkg\buildtrees\bullet3\src\3.17-e90f5ef08b.clean\src\BulletInverseDynamics\IDMath.cpp
FAILED: src/BulletInverseDynamics/CMakeFiles/BulletInverseDynamics.dir/IDMath.cpp.obj 
"C:\PROGRA~2\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\Hostx64\arm64\cl.exe"   /TP -DNO_OPENGL3 -DUSE_GRAPHICAL_BENCHMARK -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_WIN64 -IF:\vcpkg\buildtrees\bullet3\src\3.17-e90f5ef08b.clean\src /GR- /Gy /GF  /nologo /DWIN32  /W3 /utf-8  /EHsc /MP  /fp:fast /wd4244 /wd4267 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -D_DEBUG /showIncludes /Fosrc\BulletInverseDynamics\CMakeFiles\BulletInverseDynamics.dir\IDMath.cpp.obj /Fdsrc\BulletInverseDynamics\CMakeFiles\BulletInverseDynamics.dir\BulletInverseDynamics.pdb /FS -c F:\vcpkg\buildtrees\bullet3\src\3.17-e90f5ef08b.clean\src\BulletInverseDynamics\IDMath.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\emmintrin.h(24): fatal error C1189: #error:  This header is specific to X86 and X64 targe
```

--- a/ports/bullet3/portfile.cmake
+++ b/ports/bullet3/portfile.cmake
@@ -18,9 +18,8 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         multithreading       BULLET2_MULTITHREADING
 )
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
     OPTIONS
         -DUSE_MSVC_RUNTIME_LIBRARY_DLL=ON
         -DBUILD_CPU_DEMOS=OFF
@@ -32,16 +31,15 @@ vcpkg_configure_cmake(
         ${FEATURE_OPTIONS}
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/bullet TARGET_PATH share/bullet)
+vcpkg_cmake_config_fixup(PACKAGE_NAME bullet CONFIG_PATH share/bullet)
+
+vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/bullet/BulletInverseDynamics/details")
 
-vcpkg_copy_pdbs()
-
 file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
-
-vcpkg_fixup_pkgconfig()

--- a/ports/bullet3/portfile.cmake
+++ b/ports/bullet3/portfile.cmake
@@ -1,4 +1,6 @@
-vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "UWP")
+if (NOT VCPKG_TARGET_IS_OSX)
+    vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "UWP")
+endif()
 
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 

--- a/ports/bullet3/vcpkg.json
+++ b/ports/bullet3/vcpkg.json
@@ -5,6 +5,16 @@
   "description": "Bullet Physics is a professional collision detection, rigid body, and soft body dynamics library",
   "homepage": "https://github.com/bulletphysics/bullet3",
   "supports": "!((windows | linux) & (arm | uwp))",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
   "features": {
     "multithreading": {
       "description": "Multithreading functionality for bullet3"

--- a/ports/bullet3/vcpkg.json
+++ b/ports/bullet3/vcpkg.json
@@ -1,9 +1,10 @@
 {
   "name": "bullet3",
   "version": "3.17",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Bullet Physics is a professional collision detection, rigid body, and soft body dynamics library",
   "homepage": "https://github.com/bulletphysics/bullet3",
+  "supports": "!((windows | linux) & (arm | uwp))",
   "features": {
     "multithreading": {
       "description": "Multithreading functionality for bullet3"

--- a/versions/b-/bullet3.json
+++ b/versions/b-/bullet3.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "28578c32a4ac8cd357a96df13cdd8072fd9376fe",
+      "git-tree": "88add55351bf89f89e584924663861c0a827bdf5",
       "version": "3.17",
       "port-version": 3
     },

--- a/versions/b-/bullet3.json
+++ b/versions/b-/bullet3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "28578c32a4ac8cd357a96df13cdd8072fd9376fe",
+      "version": "3.17",
+      "port-version": 3
+    },
+    {
       "git-tree": "bcc42d2c8eb5fc4134c31f6176bc0c0ae3bc46e1",
       "version": "3.17",
       "port-version": 2

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1170,7 +1170,7 @@
     },
     "bullet3": {
       "baseline": "3.17",
-      "port-version": 2
+      "port-version": 3
     },
     "bustache": {
       "baseline": "1.1.0",

